src/ragger/backend/interface.py

```diff
@@ -313,3 +313,65 @@ def compare_screen_with_snapshot(self,
         :rtype: bool
         """
         raise NotImplementedError
+
+    @abstractmethod
+    def wait_for_screen_change(self, timeout: float = 10.0, context: list = []):
+        """
+        Wait until the screen content (text) changes compared to what is provided
+        by the context parameter. If no context is provided, the function returns
+        immediately, returning the current screen content.
+
+        This method may be left void on backends connecting to physical devices,
+        where a physical interaction must be performed instead.
+        This will prevent the instrumentation to fail (the void method won't
+        raise `NotImplementedError`), but the instrumentation flow will probably
+        get stuck (on further call to `receive` for instance) until the expected
+        action is performed on the device.
+
+        :param timeout: Maximum time to wait for a screen change before an
+                        exception is raised.
+        :type timeout: float
+        :param context: Context to compare screen content with.
+        :type context: list
+        :return: Latest screen content after the screen change.
+        :rtype: list
```

Comment:
```suggestion
        Wait until the screen content (text) changes compared to what is provided
        by the context parameter. If no context is provided, the function returns
        immediately, returning the current screen context.

        This method may be left void on backends connecting to physical devices,
        where a physical interaction must be performed instead.
        This will prevent the instrumentation to fail (the void method won't
        raise `NotImplementedError`), but the instrumentation flow will probably
        get stuck (on further call to `receive` for instance) until the expected
        action is performed on the device.

        :param timeout: Maximum time to wait for a screen change before an
                        exception is raised.
        :type timeout: float
        :param context: Context to compare screen content with.
        :type context: list
        :return: Latest screen context after the screen change.
        :rtype: list
```
As technically the context is opaque and may contains something else than the text content of the strng?

Reply:
See my comment here : https://github.com/LedgerHQ/ragger/pull/50#discussion_r1046954020

Reply:
Done.